src/PatrolFilter/FiltersPopover/index.js

```diff
@@ -127,7 +127,8 @@ const FiltersPopover = React.forwardRef(({
 
   const patrolLeaderFilterOptions = patrolLeaderSchema?.trackedbySchema?.properties?.leader?.enum_ext?.map(({ value }) => value)
     || [];
-  const selectedLeaders = !!selectedLeaderIds?.length ?
+
+  const selectedLeaders = !!selectedLeaderIds?.length && !isEmpty(patrolLeaderSchema) ?
```

Comment:
in the `FilterPopover` component of the patrol filters, the `selectedLeaders` prop which is passed as the value prop to `ReportedBySelect`sometimes was having this array value `[undefined]` that is because the `patrolLeaderSchema` is empty, causing the mapping of the array happens even if there is no leader information still available